axis.go

```diff
@@ -407,14 +418,63 @@ func (DefaultTicks) Ticks(min, max float64) (ticks []Tick) {
 		if val >= min && val <= max && !found {
 			ticks = append(ticks, Tick{Value: val})
 		}
-		if math.Nextafter(val, val+minorDelta) == val {
-			break
-		}
 		val += minorDelta
 	}
 	return
 }
 
+func removeDuplicates(elements []float64) []float64 {
+	// Uses a map to record duplicates as we find them.
+	encountered := map[float64]bool{}
+	result := []float64{}
+
+	for v := range elements {
+		if encountered[elements[v]] == false {
+			// Records this element as an encountered element.
+			encountered[elements[v]] = true
+			// Appends it to the result slice.
+			result = append(result, elements[v])
+		} // Otherwise, no elements are added as they are duplicates.
+	}
+	// Returns the new slice.
+	return result
+}
+
+func findRightLevelOfPrecision(elements []float64) []float64 {
+
+	var result []float64
+
+	for i := 1; i < 308; { //308 is the precision of the MaxFloat64 which we take as the maximum one here
+
+		var new_element float64
+
+		var v_mult float64
+
+		result = nil
+
+		for _, v := range elements {
+			v_mult = v * math.Pow10(i)
```

Comment:
you should probably compute `math.Pow10(i)` only once outside of the inner for-loop.
and reuse it there.

also, `v_mult` doesn't look too Go-idiomatic.
`w` ? or `vpow`, or `vp` or `v10`.

Reply:
Changed that. Thanks!